Userland/Applications/Calculator/KeypadValue.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2021, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Math.h>
+#include <AK/String.h>
+#include <AK/Types.h>
+
+class KeypadValue {
+    friend class Keypad;
+
+public:
+    KeypadValue(i64, u8);
+    KeypadValue(i64);
+
+    //helper lambda for KeypadValue's operators should have access to its private members
+    //because we have default template arguments, we need to define the lambda inside the class declaration
+    template<typename T, typename F = T(KeypadValue const&, KeypadValue const&, i64, i64, bool)>
+    ALWAYS_INLINE T operate_on_keypadvalues(KeypadValue const& lhs, KeypadValue const& rhs, F callback)
+    {
+        const KeypadValue& less_decimal_places = (lhs.m_decimal_places < rhs.m_decimal_places) ? lhs : rhs;
+        const KeypadValue& more_decimal_places = (lhs.m_decimal_places < rhs.m_decimal_places) ? rhs : lhs;
+
+        i64 more_decimal_places_equalized = more_decimal_places.m_value;
+        i64 less_decimal_places_equalized = (i64)AK::pow(10.0, (double)(more_decimal_places.m_decimal_places - less_decimal_places.m_decimal_places)) * less_decimal_places.m_value;
+
+        bool lhs_is_less = (lhs.m_decimal_places < rhs.m_decimal_places);
+
+        return callback(less_decimal_places, more_decimal_places,
+            less_decimal_places_equalized, more_decimal_places_equalized,
+            lhs_is_less);
+    }
+
+    KeypadValue operator+(KeypadValue const&);
+    KeypadValue operator-(KeypadValue const&);
+    KeypadValue operator*(KeypadValue const&);
+    KeypadValue operator-(void) const;
+    bool operator<(KeypadValue const&);
+    bool operator>(KeypadValue const&);
+    bool operator==(KeypadValue const&);
+
+    explicit KeypadValue(double);
+    explicit operator double();
+
+private:
+    i64 m_value { 0 };
+    u8 m_decimal_places { 0 };
+    //for example -123.55:
+    //m_value = -12355
+    //m_decimal_places = 2
```

Comment:
Move this comment above the members or on top of the class with some more explanation.
Also please properly format the comments according to the coding style.